specs/IFramePermissionRequested.md

```diff
@@ -0,0 +1,363 @@
+
+                                    auto cachedPermission =
+                                        m_cachedPermissions.find(cachedKey);
+                                    if (cachedPermission != m_cachedPermissions.end())
+                                    {
+                                        bool allow = cachedPermission->second;
+                                        if (allow)
+                                        {
+                                            CHECK_FAILURE(args->put_State(
+                                                COREWEBVIEW2_PERMISSION_STATE_ALLOW));
+                                        }
+                                        else
+                                        {
+                                            CHECK_FAILURE(args->put_State(
+                                                COREWEBVIEW2_PERMISSION_STATE_DENY));
+                                        }
+
+                                        PutHandled(args);
+                                        return S_OK;
+                                    }
+
+                                    std::wstring message =
+                                        L"An iframe has requested device permission for ";
+                                    message += NameOfPermissionKind(kind);
+                                    message += L" to the website at ";
+                                    message += uri.get();
+                                    message += L"?\n\n";
+                                    message += L"Do you want to grant permission?\n";
+                                    message +=
+                                        (userInitiated
+                                             ? L"This request came from a user gesture."
+                                             : L"This request did not come from a user "
+                                               L"gesture.");
+
+                                    int response = MessageBox(
+                                        nullptr, message.c_str(), L"Permission Request",
+                                        MB_YESNOCANCEL | MB_ICONWARNING);
+
+                                    COREWEBVIEW2_PERMISSION_STATE state =
+                                        COREWEBVIEW2_PERMISSION_STATE_DEFAULT;
+
+                                    if (response == IDYES)
+                                    {
+                                        m_cachedPermissions[cachedKey] = true;
+                                        state = COREWEBVIEW2_PERMISSION_STATE_ALLOW;
+                                    }
+                                    else if (response == IDNO)
+                                    {
+                                        m_cachedPermissions[cachedKey] = false;
+                                        state = COREWEBVIEW2_PERMISSION_STATE_DENY;
+                                    }
```

Comment:
The C# code has an 'else' clause here that sets state to default. We should match with that here, or remove that there.

Reply:
Removed there (set state to default before conditional)